File: frontend/lib/department.dart
class Department {
  final int id;
  String name;
  final DateTime lastModified;
  final bool synced;
  final bool newDepartment;
  final bool modified;

  Department({
    required this.id,
    required this.name,
    required this.lastModified,
    required this.synced,
    required this.newDepartment,
    required this.modified,
  });

  // Метод для создания объекта из JSON
  factory Department.fromJson(Map<String, dynamic> json) {
    return Department(
      id: json['id'],
      name: json['name'],
      lastModified: DateTime.parse(json['lastModified']),
      synced: json['synced'],
      newDepartment: json['newDepartment'],
      modified: json['modified'],
    );
  }

  // Метод для преобразования объекта в JSON
  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'name': name,
      'lastModified': lastModified.toIso8601String(),
      'synced': synced,
      'newDepartment': newDepartment,
      'modified': modified,
    };
  }
}

File: demo/src/main/java/com/example/demo/entity/Department.java
package com.example.demo.entity;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalDateTime;

@Entity
@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
public class Department {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)  // Используем автоинкремент для Long
    private Long id;  // Используем Long как уникальный идентификатор

    private String name;  // Название департамента

    private LocalDateTime lastModified;  // Дата и время последнего изменения

    private boolean synced = false;  // Флаг синхронизации с сервером

    private boolean newDepartment = false;  // Флаг для нового департамента

    private boolean modified = false;  // Флаг для изменений департамента
}